### ISSUE TITLE
[RFE] AD Provider should use tokenGroups with non-ID-mapping

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1572]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1568
- Created at 2012-10-05 18:32:49 by [sgallagh](https://pagure.io/user/sgallagh)
- Closed as Fixed
- Assigned to [pbrezina](https://pagure.io/user/pbrezina)
---

When communicating with AD providers that are using assigned POSIX IDs instead of performing automatic SID mapping, we should maintain the original SID for groups in the cache.

We want to be able to rely on cache lookups for SIDs in order to enable the use of tokenGroups lookups for fast initgroups() requests against AD.

In the first implementation, it's probably safe to assume that the POSIX ID will never change (without a full cache deletion). Thus the SID->POSIX mapping should always be correct.

### Comments

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-10-11 15:33:02](https://pagure.io/SSSD/sssd/issue/1568#229331)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.10 beta
rhbz:  => 0
summary: AD Provider should use tokenGroups with non-ID-mapping => [RFE] AD Provider should use tokenGroups with non-ID-mapping

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-12-20 22:01:06](https://pagure.io/SSSD/sssd/issue/1568#229332)**

Fields changed

design:  => 
design_review:  => 0
fedora_test_page:  => 
selected:  => Want

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-01-03 16:26:39](https://pagure.io/SSSD/sssd/issue/1568#229333)**

Fields changed

priority: major => critical

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-01-03 16:46:01](https://pagure.io/SSSD/sssd/issue/1568#229334)**

Fields changed

priority: critical => major

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-01-03 16:49:53](https://pagure.io/SSSD/sssd/issue/1568#229335)**

Fields changed

priority: major => critical

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-03-16 01:19:55](https://pagure.io/SSSD/sssd/issue/1568#229336)**

Fields changed

review:  => 1

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-04-04 16:59:29](https://pagure.io/SSSD/sssd/issue/1568#229337)**

Fields changed

owner: somebody => okos

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-04-18 14:10:02](https://pagure.io/SSSD/sssd/issue/1568#229338)**

Fields changed

status: new => assigned

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-04-19 14:25:29](https://pagure.io/SSSD/sssd/issue/1568#229339)**

Just as a reminder, please make sure to document in the manpage that the SID<->POSIX mappings require a full cache deletion if the statically-assigned POSIX ID changes in AD.

Alternately, we may want to open another ticket to extend the sss_cache tool to be able to reset the mapping state (so it's recalculated).

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-04-19 14:32:08](https://pagure.io/SSSD/sssd/issue/1568#229340)**

Replying to [comment:9 sgallagh]:
> Just as a reminder, please make sure to document in the manpage that the SID<->POSIX mappings require a full cache deletion if the statically-assigned POSIX ID changes in AD.
> 

+1

> Alternately, we may want to open another ticket to extend the sss_cache tool to be able to reset the mapping state (so it's recalculated).

Care to open a ticket? This might be a nice to have task for some external contributor..

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-05-13 15:13:06](https://pagure.io/SSSD/sssd/issue/1568#229341)**

Since there is no string change moving to 1.10.

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-05-13 15:16:49](https://pagure.io/SSSD/sssd/issue/1568#229342)**

Fields changed

milestone: SSSD 1.10 beta => SSSD 1.10.0

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-05-31 10:08:01](https://pagure.io/SSSD/sssd/issue/1568#229343)**

Fields changed

changelog:  => Performance improvement.
design:  => N/A (trivial)

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-06-13 16:52:54](https://pagure.io/SSSD/sssd/issue/1568#229344)**

Performance improvement not critical for the 1.10.0 release.

milestone: SSSD 1.10.0 => SSSD 1.10.1

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-07-18 16:46:00](https://pagure.io/SSSD/sssd/issue/1568#229345)**

Moving tickets that didn't make 1.10.1 to the 1.10.2 bucket.

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-07-18 16:49:26](https://pagure.io/SSSD/sssd/issue/1568#229346)**

Moving tickets that didn't make 1.10.1 to 1.10.2

milestone: SSSD 1.10.1 => SSSD 1.10.2

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-09-04 13:24:26](https://pagure.io/SSSD/sssd/issue/1568#229347)**

Fields changed

patch: 0 => 1

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-09-19 14:05:46](https://pagure.io/SSSD/sssd/issue/1568#229348)**

The patch is on list but since we've moved all supported releases to 1.11.x, I'd rather not add additional RFE to 1.10 only and add this enhancement to 1.11.2

milestone: SSSD 1.10.2 => SSSD 1.11.2

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-10-02 10:56:36](https://pagure.io/SSSD/sssd/issue/1568#229349)**

Fields changed

owner: okos => pbrezina
status: assigned => new

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-10-03 14:43:24](https://pagure.io/SSSD/sssd/issue/1568#229350)**

Lowering priority for 1.11.2

priority: critical => minor

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-10-25 15:28:37](https://pagure.io/SSSD/sssd/issue/1568#229351)**

Did not make 1.11.2 after all.

milestone: SSSD 1.11.2 => SSSD 1.11.3

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-12-18 13:01:46](https://pagure.io/SSSD/sssd/issue/1568#229352)**

- master: 
    - ed3e08e6ff267722c605141a0b57774efe4cb531
    - 29a61bce88147872b5086278d37b1e58726032d1

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-12-18 17:22:23](https://pagure.io/SSSD/sssd/issue/1568#229353)**

- sssd-1-11:
    - 214481d379fa2eeee2bfbe42ad813864218ee95a
    - df82d71589d62f68e3e95cb9a903d2b08d26b65a

resolution:  => fixed
status: new => closed

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2017-02-24 14:39:50](https://pagure.io/SSSD/sssd/issue/1568#229354)**

**Metadata Update from @sgallagh**:
- Issue assigned to pbrezina
- Issue marked as blocked by: #1887
- Issue set to the milestone: SSSD 1.11.3
